index.js

```diff
@@ -0,0 +1,138 @@
+const express = require("express");
+const mongoose = require("mongoose");
+const md5 = require("md5");
+const bodyParser = require("body-parser");
+const { User, access_token, address } = require("./model");
+const {
+    auth,
+    validator,
+    expiryValidator,
+} = require("./middleware/authentication");
+const { ObjectId } = require("bson");
+
+const app = express();
+app.use(express.json());
+app.use(
+    express.urlencoded({
+        extended: true,
+    })
+);
+const port = 3000 || process.env.PORT;
+const connectDb = () =>
+    new Promise((resolve, reject) => {
+        try {
+            resolve(
+                mongoose.connect(
+                    "mongodb+srv://ayush:tyagi.apj@cluster0.oix8x.mongodb.net/myFirstDatabase?retryWrites=true&w=majority",
+                    { useNewUrlParser: true, useUnifiedTopology: true }
+                )
+            );
+            console.log("connected to database");
+        } catch (err) {
+            reject(err);
+        }
+    });
+connectDb();
+app.post("/user/register", auth, async (req, res) => {
+    try {
+        if (req.body.password == req.body.conf_password) {
+            const data = {
+                username: req.body.username,
+                email: req.body.email,
+                firstname: req.body.firstname,
+                lastname: req.body.lastname,
+                password: md5(req.body.password),
+            };
+            await User.create(data);
+            res.send("usercreated");
+        } else {
+            res.send("password and confirm password did not matched");
+        }
+    } catch (er) {
+        res.send(er);
+    }
+});
+app.post("/user/login", async (req, res) => {
+    try {
+        const user = await User.findOne({ username: req.body.username });
+        if (user) {
+            const givenPassword = md5(req.body.password);
+            if (user.password == givenPassword) {
+                let data = {
+                    user_id: user._id,
+                    token: md5(Date.now()),
+                };
+                await access_token.create(data);
+                res.send(user._id);
+            } else {
+                res.status(500).send("password not matched");
+            }
+        } else {
+            res.send("user not exists");
+        }
+    } catch (er) {
+        res.send(er);
+    }
+});
+
+app.get("/user/get/:id", expiryValidator, async (req, res) => {
+    try {
+        let { token: { user_id } , params:{id}} = req;
+        if (user_id == req.params.id) {
+            const userAddress = await User.findById(user_id).populate("address")
+            res.status(200).send(userAddress)
+        }
```

Comment:
what is the use of this condition ? and 
what will happen when this if condition is false ?